File: Problems/Algorithms/1465. Maximum Area of a Piece of Cake after Horizontal and Vertical Cuts/MaxCuts.java
class Solution {
    public int maxArea(int h, int w, int[] horizontalCuts, int[] verticalCuts) {
        int MOD = 1000000007;
        
        Arrays.sort(horizontalCuts);
        
        int maxH = horizontalCuts[0];
        for (int i = 1; i < horizontalCuts.length; i++) {
            maxH = Math.max(maxH, horizontalCuts[i]-horizontalCuts[i-1]);
        }
        maxH = Math.max(maxH, h-horizontalCuts[horizontalCuts.length-1]);
        
        Arrays.sort(verticalCuts);
        
        int maxW = verticalCuts[0];
        for (int j = 1; j < verticalCuts.length; j++) {
            maxW = Math.max(maxW, verticalCuts[j]-verticalCuts[j-1]);
        }
        maxW = Math.max(maxW, w-verticalCuts[verticalCuts.length-1]);
        
        return (int) ((Long.valueOf(maxH) * Long.valueOf(maxW)) % MOD);
    }
}

File: Problems/Algorithms/1465. Maximum Area of a Piece of Cake after Horizontal and Vertical Cuts/max_cuts.rs
const MOD: i64 = 1_000_000_007;

impl Solution {
    pub fn max_area(h: i32, w: i32, horizontal_cuts: Vec<i32>, vertical_cuts: Vec<i32>) -> i32 {
        let mut hc = horizontal_cuts;
        let mut vc = vertical_cuts;
        
        hc.extend(&[0, h]);
        vc.extend(&[0, w]);
        
        hc.sort_unstable();
        vc.sort_unstable();
        
        let maxh = hc.windows(2).map(|x| x[1]-x[0]).max().unwrap() as i64;
        let maxv = vc.windows(2).map(|x| x[1]-x[0]).max().unwrap() as i64;
        
        ((maxh * maxv) % MOD) as i32
    }
}

